### PR TITLE
data URI support

### PR DESCRIPTION
**Pull request by sonnyp**
this PR includes commit for stdin support #5  as well

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -2,10 +2,15 @@
 'use strict';
 const meow = require('meow');
 const opn = require('opn');
+const getStdin = require('get-stdin');
+const tempWrite = require('temp-write');
+const dataUriToBuffer = require('data-uri-to-buffer');
```
Comment (sonnyp) [inline cli.js:7]: maybe require those when needed to reduce startup time ?

File: cli.js [hunk 2]
```diff
@@ -23,4 +30,15 @@ const cli = meow(`
...
-opn(cli.input[0], cli.flags);
+const getInput = cb => {
+	return cli.input[0] ?
+		cb(cli.input[0]) : getStdin().then(stdin => cb(stdin));
+};
+
+const transform = input => {
+	input = input.trim();
+	return input.slice(0, 5) === 'data:' ?
+		tempWrite.sync(dataUriToBuffer(input)) : input;
```
Comment (sonnyp) [inline cli.js:40]: I'm not sure if the absence of file extension will cause any issues, it seems to work good on OS X, I'll try tomorrow on Linux.

File: cli.js [hunk 3]
```diff
@@ -2,10 +2,15 @@
...
 const opn = require('opn');
+const getStdin = require('get-stdin');
+const tempWrite = require('temp-write');
+const dataUriToBuffer = require('data-uri-to-buffer');
 
 const cli = meow(`
 	Usage
 	  $ opn <file|url> [--wait] [-- <app> [args]]
+			or
+		$ stdout | opn [--wait] [-- <app> [args]]
```
Comment (sindresorhus) [inline cli.js]: wrong indentation

File: cli.js [hunk 4]
```diff
@@ -2,10 +2,15 @@
...
 const meow = require('meow');
 const opn = require('opn');
+const getStdin = require('get-stdin');
+const tempWrite = require('temp-write');
+const dataUriToBuffer = require('data-uri-to-buffer');
 
 const cli = meow(`
 	Usage
 	  $ opn <file|url> [--wait] [-- <app> [args]]
+			or
```
Comment (sindresorhus) [inline cli.js]: don't need this line

Comment (sonnyp) [inline cli.js]: fixed

File: cli.js (hunk 3 above)
Comment (sonnyp) [inline cli.js]: fixed

File: cli.js (hunk 1 above)
Comment (sindresorhus) [inline cli.js:7]: Premature optimization is the root of all evil.

http://c2.com/cgi/wiki?PrematureOptimization

A few requires have no noticeable impact.

**Comment by sindresorhus** [comment]
I think you misunderstood me. There's no need for data URI support. It can just be stdin.

```
opn data:,Hello%2C%20World!
```

Could just be

```
echo 'Hello World!' | opn
```

Note, not URL, but the content as stdin.

**Comment by sonnyp** [comment]
> I think you misunderstood me.

No 

> There's no need for data URI support.

That was my original request see https://github.com/sindresorhus/opn/issues/24

>  It can just be stdin.

Yes and the PR is at https://github.com/sindresorhus/opn-cli/pull/5

**Comment by sindresorhus** [comment]
> Yes and the PR is at #5

No, that PR adds support for reading URL from stdin. That isn't very useful as you can already do that with `opn <url>`. Instead, I asked for that it should read the data from stdin, save it to a tempfile and then open it.

**Comment by sonnyp** [comment]
@sindresorhus right my bad, understood

**Comment by sindresorhus** [comment]
It will probably need an `--ext` flag too to indicate the filetype.

We could default to `.txt` for text and detect binary with https://github.com/sindresorhus/file-type, so it would be automatic in most cases.

**Comment by sindresorhus** [comment]
Closing in favor of #7.

**Pull request closed without merging**